test-result-summary-client/src/TabularView/TabularView.jsx

```diff
@@ -0,0 +1,718 @@
+    onBenchmarkChange = value => {
+        this.setState({benchmarkFilter: value});
+        this.benchmarkFilter(value);
+    }
+    // Date picker format needs to be converted to match the one available in the database entry
+    dateTransform(date, type) {
+        const dateSplit = date.split('/');
+        // Database date format: YYYYMMDD
+        const jdkDate = dateSplit[2] + ((dateSplit[0].length < 2) ? "0" + dateSplit[0] : dateSplit[0]) + ((dateSplit[1].length < 2) ? "0" + dateSplit[1] : dateSplit[1]);
+
+        if (type === 'test') {
+            this.setState({testJdkDate: jdkDate});
+        } else {
+            this.setState({baselineJdkDate: jdkDate});
+        }
+    }
+    // Based on data returned from database, create relevant columns for platforms
+    generateColumns(platforms) {
+        const newArray = [];
+        let column = {
+        Header: 'Benchmark Name',
+        accessor: 'benchmarkNVM',
+        // The three line breaks ensure the benchmark name, variant and metric appear in separate lines
+        Cell: props => <span>{props.value.split(",")[0]} <br/> {props.value.split(",")[1]} <br/> {props.value.split(",")[2]}</span>
+        };
+
+        newArray.push(column);
+
+        for (let i = 0; i < platforms.length; i++) {
+            let platform = platforms[i];
+    	
+            column = {};
+            column.id = platform;
+            column.Header = platform.toUpperCase();
+            column.accessor = d => d.platformsSpecificData[platform];
+            // Each cell needs to display the comparison by default, on hover displays further details
+            column.Cell = props => <Tooltip title={<div >
+                Raw Test Score: {this.handleProp(props.value, 'testScore')} <br/> 
+                Test CI: {this.handleProp(props.value, 'testCI')}  <br/>
+                Test JDK: {this.handleProp(props.value, 'testJdk')}	<br/> 
+                Test Sdk Resource: {this.handleProp(props.value, 'testSdkResource')}	<br/> 
+                Raw Baseline Score: {this.handleProp(props.value, 'baselineScore')} <br/> 
+                Basline CI: {this.handleProp(props.value, 'baselineCI')}  <br/>
+                Baseline JDK: {this.handleProp(props.value, 'baselineJdk')} <br/>
+                Baseline Sdk Resource: {this.handleProp(props.value, 'baselineSdkResource')}
+                </div> }>
+                <span onClick={() => this.handleLink(this.handleProp(props.value, 'buildUrl'))}> {this.handleProp(props.value, 'compare')} % </span></Tooltip>;
+            column.getProps = (state, rowInfo) => (this.handleRegression(this.handleProp(rowInfo.row[platform], 'compare')));
+            newArray.push(column);
+        }
+        this.setState({columns:newArray, originalColumns: newArray});
+    }
+    // Set cell color based on comparison value
+    handleRegression (val) {
+        let color;
+        if (val === 0) {
+            color = '#ffdbac';}
+        else if (val <= 90) {
```

Comment:
We should use both upper and lower ranges just to make it more robust. Otherwise, we're relying on the order of if else statements.